lib/app/service/pull.go

```diff
@@ -407,9 +435,12 @@ type pullState struct {
 	packages map[loc.Locator]struct{}
 }
 
-// IsMetadataPackage determines if the specified package is a metadata package.
-// A metadata package describes a remote package and deserves
-// special handling in certain cases.
-func IsMetadataPackage(envelope pack.PackageEnvelope) bool {
-	return envelope.RuntimeLabels[pack.PurposeLabel] == pack.PurposeMetadata
+// copyRuntimeLabels copies labels from runtimeLabels to labels for each
+// label non-existent in labels
+func copyRuntimeLabels(labels map[string]string, runtimeLabels map[string]string) {
```

Comment:
Nit: `(labels, runtimeLabels map[string]string)`. Also, maybe `srcLabels` and `dstLabels` would be a bit more suitable names?